### DISCUSSION TITLE
Syntax for hex colors

### DISCUSSION THREAD

**Discussion by fabd** [windicss/windicss#125] [Ideas]
I was looking at the syntax for hex colors:

`bg-hex-224466`

SInce utility classes are quite verbose in templates, would it make sense to use:

`bg-#224466`

I like this syntax because visually it immediately pops out as a color, where as `bg-hex-fc0020` not quite. It's shorter which is helpful in the very verbose tailwind style templates. When doing search/replace in legacy codebase it might be handy to be able to match with the dash, to avoid any potential other matches (eg. search for `#224466` instead of `224466`), this would then match across legacy css / scss and the windicss utilities.

Is there a way to create a utility like that through config?

Great job by the way I dreamed of something like this a couple months ago. Was pretty stoked to see someone made it!

**Comment by voorjaar** [comment]
You can simply add a plugin.

```js
plugins: [
  plugin(function ({ addDynamic }) {
    addDynamic('bg', ({ Utility, Style, Property }) => {
      if (/^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/.test(Utility.body)) {
        return Style(Utility.class, [
          Property('--tw-bg-opacity', '1'),
          Property('background-color', `rgba(${hex2RGB(Utility.body)?.join(', ')}, var(--tw-bg-opacity))`),
        ]);
      }
    });
  }),
],
```

But we will not change syntax for now. `hex` may be more readable to me, and we also plan to change `bg-$test-variable` to `bg-var-test-variable`.

**Reply by ahku** [reply]
This might be a sensitive topic, but would copying the offical bracket syntax be out of question? Like `bg-[#ff000]`. It's quite nice to immediately see where the exceptions are in your html.

Anyway, I see advantages for both approaches: `bg-hex-` might be better if you want to support `bg-hsl` later, but if you're want to be consistent, `bg-#` might make a bit more sense since you support unit values such as `pt-1px`, `pt-1rem`. Everybody understands that `#ff000` or `#f00` is a hex color value.

**Reply by voorjaar** [reply]
Changing the grammar is a very dangerous thing, because someone is using the previous grammar, so switching will always bring a lot of problems. Therefore, we will not do this as a last resort, or I will consider it when the major version is updated.
